src/interpreter.rs

```diff
@@ -120,3 +129,1207 @@ bitflags::bitflags! {
+
+                            if op == OP_NUMEQUALVERIFY {
+                                if cast_to_bool(stack.top(-1)?) {
+                                    stack.pop()?;
+                                } else {
+                                    return set_error(ScriptError::NumEqualVerify);
+                                }
+                            }
+                        }
+
+                        OP_WITHIN => {
+                            // (x min max -- out)
+                            if stack.size() < 3 {
+                                return set_error(ScriptError::InvalidStackOperation);
+                            }
+                            let bn1 = ScriptNum::new(stack.top(-3)?, require_minimal, None)?;
+                            let bn2 = ScriptNum::new(stack.top(-2)?, require_minimal, None)?;
+                            let bn3 = ScriptNum::new(stack.top(-1)?, require_minimal, None)?;
+                            let value = bn2 <= bn1 && bn1 < bn3;
+                            stack.pop()?;
+                            stack.pop()?;
+                            stack.pop()?;
+                            stack.push_back(if value {
+                                vch_true.clone()
+                            } else {
+                                vch_false.clone()
+                            })
+                        }
+
+                        //
+                        // Crypto
+                        //
+                        OP_RIPEMD160
+                        | OP_SHA1
+                        | OP_SHA256
+                        | OP_HASH160
+                        | OP_HASH256 => {
+                            // (in -- hash)
+                            if stack.size() < 1 {
+                                return set_error(ScriptError::InvalidStackOperation);
+                            }
+                            let vch = stack.top(-1)?;
+                            let mut vch_hash = vec![];
+                            if op == OP_RIPEMD160 {
+                                vch_hash = Ripemd160::digest(vch).to_vec();
+                            } else if op == OP_SHA1 {
+                                let mut hasher = Sha1::new();
+                                hasher.update(vch);
+                                vch_hash = hasher.finalize().to_vec();
+                            } else if op == OP_SHA256 {
+                                vch_hash = Sha256::digest(vch).to_vec();
+                            } else if op == OP_HASH160 {
+                                vch_hash = Ripemd160::digest(Sha256::digest(vch)).to_vec();
+                            } else if op == OP_HASH256 {
+                                vch_hash = Sha256::digest(Sha256::digest(vch)).to_vec();
+                            }
+                            stack.pop()?;
+                            stack.push_back(vch_hash)
+                        }
+
```

Comment:
I got to here in my review before my eyes started glazing over. Will resume probably on Monday.